### PR TITLE
feat(directive): add validation and model value formatting

### PR DESCRIPTION
**Pull request by bighappyface**
This PR adds validation and model value assignment/formatting to the directive.

### CODE REVIEW

**Comment by bighappyface** [comment]
@SyntaxStacks please review

**Comment by bighappyface** [comment]
@SyntaxStacks I have modified the Travis build to run each step of the CI process as separate scripts. This allows us to fail the build for any given step whereas the `ci.sh` script would only fail the build if the last command didn't exit with 0.

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,15 @@ node_js:
   - "0.11"
   - "0.10"
   - "iojs"
+install:
```
Comment (SyntaxStacks) [inline .travis.yml]: Very cool. I'm familiar with the job that travis does but the configuration of travis is awesome.

**Comment by bighappyface** [comment]
@SyntaxStacks rebase and CI build are complete

**Pull request merged**